pyiron_base/generic/hdfio.py

```diff
@@ -31,6 +31,36 @@
 __status__ = "production"
 __date__ = "Sep 1, 2017"
 
+_special_symbol_replacements = {
+    '.': 'd',
+    '-': 'm',
+    '+': 'p',
+    ',': 'c',
+    ' ': '_',
+}
+
+
+def _get_safe_job_name(name, ndigits=8, extension=None):
+    def round_(value, ndigits=ndigits):
+        if isinstance(value, float) and ndigits is not None:
+            return round(value, ndigits=ndigits)
+        return value
+    if not isinstance(name, str):
+        name_rounded = [round_(nn) for nn in name]
+        job_name = '_'.join([str(nn) for nn in name_rounded])
+    else:
+        job_name = name
+    if extension is not None and job_name.endswith(extension):
+        job_name = job_name[:job_name.rfind(extension)]
+    job_name = job_name.split('/')
+    for k, v in _special_symbol_replacements.items():
+        job_name[-1] = job_name[-1].replace(k, v)
+    job_name = '/'.join(job_name)
+    if extension is not None:
+        job_name += extension
+    return job_name
```

Comment:
I just found that we only allow valid python identifiers with a length<50 for job names and that we have a function to check this. Thus, I suggest to use `from pyiron_base.job.util import _is_valid_job_name` and end the function with 

```suggestion
    try:
        _is_valid_job_name(job_name)
    except ValueError as e:
        raise ValueError(f"Could not convert {name} to valid job name"} from e
    else:
        return job_name
```

Reply:
Aaaaaaah good point